acados/sim/sim_common.c

```diff
@@ -275,16 +289,18 @@ acados_size_t sim_out_calculate_size(void *config_, void *dims)
 
     acados_size_t size = sizeof(sim_out);
 
-    int nx, nu, nz;
+    int nx, nu, nz, np = 0;
     config->dims_get(config_, dims, "nx", &nx);
     config->dims_get(config_, dims, "nu", &nu);
     config->dims_get(config_, dims, "nz", &nz);
+    config->dims_get(config_, dims, "np", &np);
 
     int NF = nx + nu;
     size += sizeof(sim_info);
 
     size += nx * sizeof(double);                // xn
     size += nx * NF * sizeof(double);           // S_forw
+    size += nx * np * sizeof(double);       // S_p (parameter sens, may be 0)
```

Comment:
Most new memory assignments and all potentially costly operations should be conditional on the option: `sens_forw_p`.
In particular, since often np >> nx, assigning nx * np memory can increase the memory footprint significantly.

Here in `sim_out`, there is the issue that the assign functions do not depend on `opts`.
I would mitigate this, by not including `S_p` in `sim_out` and instead directly get it from the integrator memory `sim_erk_memory_get`.